### ISSUE TITLE
UWP Package Manifest Vid/Pid Issue

### ISSUE THREAD

**Issue by MelbourneDeveloper** [MicrosoftDocs/winrt-api#749]
In the UWP package manifest, you must specify the Vid and Pid of a device you wish to connect to. I have been successful in getting this to work with one Vid, but some Vids are not allowed because they fail the validation. This means that some Hid devices cannot be connected to. Please see these issues: https://github.com/tpn/winsdk-10/issues/2, https://github.com/MelbourneDeveloper/KeepKey.Net/issues/2 , https://stackoverflow.com/questions/51219062/uwp-usb-hid-device-wont-connect, and 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 83152a6e-786e-58ea-7af9-71046fbdcce4
* Version Independent ID: f1ddf84f-a0a6-f20b-e7e2-c3842b24e69f
* Content: [HidDevice Class (Windows.Devices.HumanInterfaceDevice) - UWP app developer](https://docs.microsoft.com/en-us/uwp/api/windows.devices.humaninterfacedevice.hiddevice)
* Content Source: [https://cpubwin.visualstudio.com/DefaultCollection/windows-uwp/_git/winrt-api-build?path=%2fwinrt-api-build%2fwindows.devices.humaninterfacedevice.hiddevice.yml&version=GBlive&_a=contents](https://cpubwin.visualstudio.com/DefaultCollection/windows-uwp/_git/winrt-api-build?path=%2fwinrt-api-build%2fwindows.devices.humaninterfacedevice.hiddevice.yml&version=GBlive&_a=contents)
* Product: **uwp**
* Git Login: **wdg-dev-content**
* Microsoft Alias: **wdg-dev-content**

**Comment by MelbourneDeveloper** [comment]
@ginami . Hi. Glad to see someone is looking at this. It's holding up my software in a big way. My app Hardfolio (https://www.microsoft.com/en-au/p/hardfolio/9p8xx70n5d2j) is a crypto currency app for hardwarewallets. I currently can't support the most popular hardwarewallet Ledger (https://www.ledger.com/) because of this problem.

**Comment by mcleanbyron** [comment]
Hi @MelbourneDeveloper - Thanks for filing this issue. I created a new UWP project in Visual Studio 2017 targeting version 1809, opened up the Package.appxmanifest (Open -> View Code), and copied the following exact XML into the default <Capabilities> element, based this on your [stackoverflow thread](https://stackoverflow.com/questions/51219062/uwp-usb-hid-device-wont-connect).

```
<DeviceCapability Name="humaninterfacedevice">
  <Device Id="vidpid:2C97 0001">
    <Function Type="usage:0005 *" />
    <Function Type="usage:FF00 0001" />
    <Function Type="usage:ff00 *" />
  </Device>
</DeviceCapability>
```

I do not see a validation error, and can build the project without problem.

Strangely though, when I try copying your entire <DeviceCapability> element directly from your [stackoverflow thread](https://stackoverflow.com/questions/51219062/uwp-usb-hid-device-wont-connect) into the manifest, then I *do* get a validation error. The characters are the same, but for some reason copying it out of your issue page seems to introduce the validation error. However, if I then delete the "vidpid:2C97 0001" string and then manually retype this string myself, I resolve the validation error.

Can you confirm whether you can resolve the validation error via either of the options I presented above? I have no idea what is going on at this point and this doesn't appear to be a documentation issue (this repo is specifically about feedback on the docs), so if this issue persists for you I'll probably need to refer you to [contact support](https://developer.microsoft.com/en-us/windows/support).

**Comment by MelbourneDeveloper** [comment]
@mcleanbyron , thanks for getting back to me. 

> Can you confirm whether you can resolve the validation error via either of the options I presented above?

Sorry, what option should I try?

Do you mean typing it in manually? I've tried that but it did not resolve the problem.

> if this issue persists for you I'll probably need to refer you to contact support.

Please, for the love of God, do not refer me to support. This is a bug in some aspect of the framework somewhere. Please help me identify the team responsible for dealing with this so that they can recognize the problem and fix it.

**Comment by MelbourneDeveloper** [comment]
@mcleanbyron this is clearly a bug. I have tried pasting in the XML as you suggested and it doesn't help. Could you please help me find the team that is responsible for this? I logged a bug a long time ago with the repo where this XML schema sits but I haven't heard anything. This is a serious problem for Hid developers and I need to get on to the right person at Microsoft.

**Comment by mcleanbyron** [comment]
Hi @MelbourneDeveloper - I see your edited comments from several days ago where you confirmed that the steps I suggested aren't working for you. I'm sorry you're running into this issue, and I wish I could help you more but this isn't the proper venue for technical assistance. This GitHub repo is specifically for feedback/problems with the documentation, and this does not seem to be a problem with the docs for the HidDevice class. Since you are reporting a technical issue instead of a problem with the docs, I'm closing this issue for now. 

I see that you opened a thread on the Windows developer forum [here](https://social.msdn.microsoft.com/Forums/en-US/8b8499c8-b5f7-4af6-8891-ef8b8b16f989/where-can-i-log-this-bug?forum=wpdevelop) and a Microsoft support engineer is engaging with you on that thread. For what it's worth, I can confirm that I'm not seeing this validation error with the XML snippet I provided above in Visual Studio 2017 version 15.8.9. 

Thanks for reaching out, and best of luck with your issue.
